### ISSUE TITLE
[astrometry-net]: Astrometry.net package's bottle is built without netpbm support

### ISSUE THREAD

**Issue by dstndstn** [Homebrew/homebrew-core#129721]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
> brew config
HOMEBREW_VERSION: 4.0.15-97-ge191b82
ORIGIN: https://github.com/Homebrew/brew
HEAD: e191b827cce991f52be36b0a8fc3daed3ef66143
Last commit: 30 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: dacc6a270ac92756285093103c17c1dc2a31b6f4
Core tap last commit: 3 months ago
Core tap branch: master
Core tap JSON: 29 Apr 00:54 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.3UgCSmMvll/org.xquartz:0
HOMEBREW_EDITOR: emacs -nw
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.36.1 => /usr/local/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.3-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2

> brew doctor
Your system is ready to brew.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I'm the upstream maintainer.

I was trying to use the "solve-field" program to recognize an astronomical image (ie, astrometry-net's core purpose) in PNG or JPEG format.  Under the hood, this is done by converting to PNM and then using netpbm to read the image.  But netpbm is optional, so if it is missing, some additional features like debugging plots are disabled.

It looks like the bottled version of astrometry.net does NOT have netpbm (even though the formula tries to set it up).

### What happened (include all command output)?

```
> solve-field --continue -v /tmp/AlphaCamelopardis_s1024.png 2>&1 | grep -A 3 plotxy
Running: /usr/local/bin/plotxy -I /tmp/tmp.ppm.Milka2 -i /tmp/AlphaCamelopardis_s1024.axy -C red -w 2 -N 50 -x 1 -y 1 -P | /usr/local/bin/plotxy -i /tmp/AlphaCamelopardis_s1024.axy -I - -w 2 -r 3 -C red -n 50 -N 200 -x 1 -y 1 > /tmp/AlphaCamelopardis_s1024-objs.png
cairoutils.c:715:cairoutils_read_ppm_stream: Netpbm is not available; can't read PPM images
cairoutils.c:715:cairoutils_read_ppm_stream: Netpbm is not available; can't read PPM images
cairoutils.c:715:cairoutils_read_ppm_stream: Netpbm is not available; can't read PPM images
```
and it did not create the debugging plots.

### What did you expect to happen?

I expected it to produce debugging plots ([INPUT]-objs.png, specifically).

(Any JPEG or PNG input image should work for reproducing the error.)

Under the hood, I expect the `plot-xy` executable to be linked against `libnetpbm`, but it isn't:

```
> otool -L /usr/local/bin/plotxy
/usr/local/bin/plotxy:
	/usr/local/opt/gsl/lib/libgsl.27.dylib (compatibility version 28.0.0, current version 28.0.0)
	/usr/local/opt/gsl/lib/libgslcblas.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/usr/local/opt/wcslib/lib/libwcs.7.dylib (compatibility version 7.0.0, current version 7.12.0)
	/usr/local/opt/cairo/lib/libcairo.2.dylib (compatibility version 11603.0.0, current version 11603.0.0)
	/usr/local/opt/libpng/lib/libpng16.16.dylib (compatibility version 56.0.0, current version 56.0.0)
	/usr/local/opt/jpeg-turbo/lib/libjpeg.8.dylib (compatibility version 8.0.0, current version 8.2.2)
```

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
Now, if I reinstall from source with verboseness turned up, I can see that it is failing to find the libnetpbm library:

> brew reinstall -s -vv astrometry-net
[...]
Testing netpbm...
   NETPBM_INC is -I/usr/local/opt/netpbm/include/netpbm
   NETPBM_LIB is -L/usr/local/opt/netpbm/lib -lnetpbm
((clang -o os-features-test-netpbm \
...
 -L/usr/local/opt/netpbm/lib -lnetpbm >> os-features.log && \
  ./os-features-test-netpbm >> os-features.log && \
  echo "#define HAVE_NETPBM 1") \
|| echo "#define HAVE_NETPBM 0") >> ../include/astrometry/os-features-config.h.tmp
ld: library not found for -lnetpbm
clang: error: linker command failed with exit code 1 (use -v to see invocation)

[...]
```

and I can see that, indeed, 
```
> pkg-config --libs netpbm
-L/usr/local/Cellar/netpbm/10.86.38_1/lib -lnetpbm
```
but that directory does not contain a "plain old" libnetpbm.dylib, only versioned files,
```
> ls -1 /usr/local/Cellar/netpbm/10.86.38_1/lib
libnetpbm.11.86.dylib
libnetpbm.11.dylib
pkgconfig
```

I can fix this, if I first symlink the netpbm library:
```
cd /usr/local/opt/netpbm/lib
ln -s libnetpbm.11.dylib libnetpbm.dylib
```
and then reinstall astrometry-net from source:
```
> brew reinstall astrometry-net -s -vv
```

Then it works:
```
> solve-field --continue -v /tmp/AlphaCamelopardis_s1024.png 2>&1 | grep -A 3 plotxy
Running: /usr/local/bin/plotxy -I /tmp/tmp.ppm.L6JFwt -i /tmp/AlphaCamelopardis_s1024.axy -C red -w 2 -N 50 -x 1 -y 1 -P | /usr/local/bin/plotxy -i /tmp/AlphaCamelopardis_s1024.axy -I - -w 2 -r 3 -C red -n 50 -N 200 -x 1 -y 1 > /tmp/AlphaCamelopardis_s1024-objs.png
Solving...
Running:
  /usr/local/bin/astrometry-engine --verbose /tmp/AlphaCamelopardis_s1024.axy
```
and plotxy is linked to netpbm as expected:
```
> otool -L /usr/local/bin/plotxy
/usr/local/bin/plotxy:
	/usr/local/opt/gsl/lib/libgsl.27.dylib (compatibility version 28.0.0, current version 28.0.0)
	/usr/local/opt/gsl/lib/libgslcblas.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/usr/local/opt/wcslib/lib/libwcs.7.dylib (compatibility version 7.0.0, current version 7.12.0)
	/usr/local/opt/cairo/lib/libcairo.2.dylib (compatibility version 11603.0.0, current version 11603.0.0)
	/usr/local/opt/libpng/lib/libpng16.16.dylib (compatibility version 56.0.0, current version 56.0.0)
	/usr/local/opt/netpbm/lib/libnetpbm.11.86.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/local/opt/jpeg-turbo/lib/libjpeg.8.dylib (compatibility version 8.0.0, current version 8.2.2)
```

SO, long story short, I think the solution is some combination of:

1. netpbm formula should create the libnetpbm.dylib link?  Or run some mac process that produces it?

2. netpbm's pkg-config entry should use the versioned library name?

3. the astrometry-net formula should use netpbm's pkg-config rather than hard-coding NETPBM_INC and NETPBM_LIB environment variables?

4. the astrometry-net formula should point to the versioned netpbm library?
```

**Comment by carlocab** [comment]
Thanks for mailing the problem down; that's very helpful.

Let me think about the right way to handle this and then get this fixed.

**Comment by dstndstn** [comment]
Thank you very much!!
